lib/core/src/display/rendering/html/html_renderer.rs

```diff
@@ -0,0 +1,185 @@
+use crate::prelude::*;
+
+use crate::display::rendering::GraphicsRenderer;
+use crate::display::rendering::Scene;
+use crate::display::rendering::Camera;
+use crate::display::rendering::html::HTMLObject;
+use crate::math::utils::IntoFloat32ArrayView;
+use crate::math::utils::eps;
+use crate::math::utils::invert_y;
+use crate::system::web::Result;
+use crate::system::web::create_element;
+use crate::system::web::dyn_into;
+use crate::system::web::NodeInserter;
+use crate::system::web::StyleSetter;
+
+use nalgebra::Vector2;
+use nalgebra::Matrix4;
+
+use wasm_bindgen::prelude::wasm_bindgen;
+use wasm_bindgen::JsValue;
+use web_sys::HtmlElement;
+
+mod js {
+    use super::*;
+    #[wasm_bindgen(module = "/js/html_renderer.js")]
+    extern "C" {
+        pub fn set_object_transform(dom: &JsValue, matrix_array: &JsValue);
+        pub fn setup_perspective(dom: &JsValue, znear: &JsValue);
+        pub fn setup_camera_transform
+        ( dom          : &JsValue
+        , znear        : &JsValue
+        , half_width   : &JsValue
+        , half_height  : &JsValue
+        , matrix_array : &JsValue
+        );
+    }
+}
+
+fn set_object_transform(dom: &JsValue, matrix: &Matrix4<f32>) {
+    // Note [Unsafe Float32ArrayView]
+    unsafe {
+        let matrix_array =  matrix.into_float32_array_view();
+        js::set_object_transform(&dom, &matrix_array);
+    }
+}
```

Comment:
In such case, after refactoring it ot separate function, its IMO better to put the documentation + Safety section directly on the function `set_object_transform`. The whole point of refactoring it this way is that we are just encapsulating unsafe things in a safe interface, so it should be documented here. There is no more need to do some redirection - and thats super cool, cause we dont need to search for the docs in the file anymore! :)

Reply:
Ah, yes! But the only reason I am redirecting is because I am using the same artifice in `setup_camera_transform`, so I didn't want to have a duplicate comment. Makes sense?

Reply:
I would copy-paste it. Unsafe blocks are rare and are extremaly dangereous. Its error prone to refaactoring. Lets keep safety notes close to the source.